src/Modal/index.jsx

```diff
@@ -178,30 +188,35 @@ class Modal extends React.Component {
             [styles.fade]: !open,
           })}
           role="presentation"
-          onClick={this.close}
         />
         <div
-          className={classNames(
-            styles.modal,
-            {
-              [styles['modal-dialog']]: open,
-              [styles['d-block']]: open,
-              [styles.show]: open,
-              [styles.fade]: !open,
-              [styles['is-ie11']]: this.isIE11,
-            },
-          )}
-          role="dialog"
-          aria-modal
-          aria-labelledby={this.headerId}
-          {...(!renderHeaderCloseButton ? { tabIndex: '-1' } : {})}
-          {...(!renderHeaderCloseButton ? { ref: this.setFirstFocusableElement } : {})}
+          id="modal"
+          className={classNames(styles.modal, {
+            [styles.show]: open,
+            [styles.fade]: !open,
+            [styles['d-block']]: open,
+            [styles['is-ie11']]: this.isIE11,
+          })}
+          role="presentation"
+          onClick={e => this.close(e)}
         >
-          <div className={styles['modal-content']}>
-            <div className={styles['modal-header']}>
-              <h2 className={styles['modal-title']} id={this.headerId}>{this.props.title}</h2>
-              { renderHeaderCloseButton &&
+          <div
+            className={classNames({
+              [styles['modal-dialog']]: open,
+            })}
+            styles={{ zIndex: '2000' }}
+            role="dialog"
+            aria-modal
+            aria-labelledby={this.headerId}
+            {...(!renderHeaderCloseButton ? { tabIndex: '-1' } : {})}
+            {...(!renderHeaderCloseButton ? { ref: this.setFirstFocusableElement } : {})}
+          >
+            <div className={styles['modal-content']}>
+              <div className={styles['modal-header']}>
+                <h2 className={styles['modal-title']} id={this.headerId}>{this.props.title}</h2>
+                { renderHeaderCloseButton &&
                 <Button
+                  id="paragonHeaderCloseModalButton"
```

Comment:
Do we have to use hard-coded ids? It makes me slightly nervous given that ids need to be unique.